chia/wallet/wallet_node.py

```diff
@@ -75,6 +75,21 @@ def __init__(self):
         self.block_requests = {}
         self.states_validated = {}
 
+    def clear_after_height(self, height: int):
+        # Remove any cached item which relates to an event that happened at a height above height.
+        self.blocks = {k: v for k, v in self.blocks.items() if k <= height}
+        self.block_requests = {k: v for k, v in self.block_requests.items() if k[0] <= height and k[1] <= height}
+        self.ses_requests = {k: v for k, v in self.ses_requests.items() if k <= height}
+
+        remove_keys_states: List[bytes32] = []
+        for k4, coin_state in self.states_validated.items():
+            if coin_state.created_height is not None and coin_state.created_height > height:
+                remove_keys_states.append(k4)
+            elif coin_state.spent_height is not None and coin_state.spent_height > height:
+                remove_keys_states.append(k4)
+        for k5 in remove_keys_states:
+            self.states_validated.pop(k5)
```

Comment:
This suggestion is incorrect, we are not checking if `spent_height is None` before accessing the value. This is why I thought it was better in traditional form, because it's a lot of checks to fit into a comprehension.

Reply:
Oops, yes, I mis-read both as `.created_height`.  I think having consistent filter-existing vs. create-new behavior across the elements would be good even if using a loop to create the new object.

Reply:
Yes good point